bff/pkg/server/server.go

```diff
@@ -319,67 +339,72 @@ func metrics_handler(route_configs map[string]RouteConfig, dbIdentifiers []strin
 	})
 }
 
-func activity_handler(metrics_service metrics.Service) http.HandlerFunc {
+type ActivityParams struct {
+	DbIdentifier string `validate:"required,dbIdentifier,max=50"`
+	DatabaseList string `validate:"required,databaseList,max=50"`
+
+	Start string `validate:"required"`
+	End   string `validate:"required,afterStart"`
+	Step  string `validate:"required,duration"`
+
+	Legend            string `validate:"required,dim"`
+	Dim               string `validate:"required,dim"`
+	FilterDim         string `validate:"omitempty,dim"`
+	FilterDimSelected string `validate:"omitempty,filterDimSelected"`
+
+	Limit       string `validate:"omitempty,numeric,gt=0"`
+	LimitLegend string `validate:"omitempty,numeric,gt=0"`
+}
+
+func activity_handler(metrics_service metrics.Service, validate *validator.Validate) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Println("Handling activity request")
 
-		database_list := r.URL.Query().Get("database_list")
-		start := r.URL.Query().Get("start")
-		end := r.URL.Query().Get("end")
-		step := r.URL.Query().Get("step")
-		legend := r.URL.Query().Get("legend")
-		dim := r.URL.Query().Get("dim")
-		filterdim := r.URL.Query().Get("filterdim")
-		filterdimselected := r.URL.Query().Get("filterdimselected")
-		limitDim := r.URL.Query().Get("limitdim")
-		limitLegend := r.URL.Query().Get("limitlegend")
-		dbIdentifier := r.URL.Query().Get("dbidentifier")
-
-		requiredParamMap := map[string]string{
-			"dbidentifier":  dbIdentifier,
-			"database_list": database_list,
-			"start":         start,
-			"end":           end,
-			"step":          step,
-			"legend":        legend,
-			"dim":           dim,
+		params := ActivityParams{
+			DbIdentifier:      r.URL.Query().Get("dbidentifier"),
+			DatabaseList:      r.URL.Query().Get("database_list"),
+			Start:             r.URL.Query().Get("start"),
+			End:               r.URL.Query().Get("end"),
+			Step:              r.URL.Query().Get("step"),
+			Legend:            r.URL.Query().Get("legend"),
+			Dim:               r.URL.Query().Get("dim"),
+			FilterDim:         r.URL.Query().Get("filterdim"),
+			FilterDimSelected: r.URL.Query().Get("filterdimselected"),
+			Limit:             r.URL.Query().Get("limitdim"),
+			LimitLegend:       r.URL.Query().Get("limitlegend"),
 		}
 
-		for paramName, paramValue := range requiredParamMap {
-			if paramValue == "" {
-				http.Error(w, fmt.Sprintf("Missing param/value: %s", paramName), http.StatusBadRequest)
-				return
-			}
+		if err := validate.Struct(params); err != nil {
+			fmt.Println("Validation failed: ", err)
```

Comment:
Let's use `log.Printf` instead:
```suggestion
			log.Printf("Validation failed: %w", err)
```

Reply:
It's even better if you wrap it inside `if cfg.Debug { ... }`
```suggestion
			if cfg.Debug {
				log.Printf("Validation failed: %w", err)
			}
```